.github/workflows/deploy-distributables.yml

```diff
@@ -9,6 +9,8 @@ on:
 jobs:
```

Comment:
Please add top-level `permissions:`.

Reply:
`permissions:` in top-level will cover permission all jobs, And i think no need set `permission` per job when `top-level` already write exclude job `check-version` must set `permission: read`. Agree?

Reply:
I think generally we should set top-level `permissions:` read only.
Top level `permissions:` is the default, and it can be override in jobs.

I don't know the default setting of this organization. So if it is read write permissions, jobs without permission can write. It is not safe to supply chain attacks.

Reply:
You can set default permission in organization, open `https://github.com/organizations/codeigniter4/settings/actions` and find `Workflow permissions` in bottom

Reply:
I have no permission to access the URL.

Reply:
![image](https://user-images.githubusercontent.com/97607754/209750360-f4e8c3fa-1023-4a90-9ce8-baa7e16c7204.png)

Maybe @MGatner can access this setting like above?

Reply:
Yes.

Reply:
or only owner this organization can access this setting

Reply:
Yes I have access, though I'm hesitant to make changes until we have the necessary write permissions in place everywhere first. 

![image](https://user-images.githubusercontent.com/17572847/209949723-4d538fe4-aa81-4579-ba88-68e122fe55c4.jpeg)

Reply:
Default org permission should be read-only. Adding write permissions should be done on a per-workflow or per-job basis only.

Reply:
Yes, said @paulbalandan, and we must not forget that one day if we want to change permissions at the organizational level, we must include permission read at the top-level.

Btw, i am @ddevsr. Waiting review recovery account.